test/upgrade/olm_upgrade_test.go

```diff
@@ -0,0 +1,315 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/project-codeflare/codeflare-operator/test/e2e"
+	. "github.com/project-codeflare/codeflare-operator/test/support"
+)
+
+var (
+	namespaceName      = "test-ns-olmupgrade"
+	serviceAccountName = "sdk-user"
+)
+
+// Creates a Ray cluster
+func TestMNISTRayClusterUp(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+
+	// Create a namespace
+	ns := NewNamespace(namespaceName)
+	namespace := ns.CreateTestNamespaceWithName(test, namespaceName)
```

Comment:
Is there any reason for having to declare the `Namespace` struct?
Wouldn't it be enough to pass only namespace name, which is already declared in this file?
i.e.:
```suggestion
	namespace := CreateTestNamespaceWithName(test, namespaceName)
```

Reply:
Yes its not required here , i refactored to declare the Namespace struct for the test TestMnistJobSubmit and adjusted here aswell..

Reply:
yes, but even in the `TestMnistJobSubmit` you can use just the namespace name declared as namespaceName variable in this file:
`namespace := GetNamespaceWithName(test, namespaceName)`
there is no need to create a dedicated struct for namespace as all the information you need you already have here

Reply:
Yeah right, will remove struct and update the same